File: 백준/Bronze/2231. 분해합/분해합.py
import sys
def print(*args, sep=" ", end="\n"):
    sys.stdout.write(sep.join(map(str, args)) + end)
def input():
    return sys.stdin.readline().rstrip()

def main():
    n_str = input()
    n = int(n_str)
    start = max(0, int(n) - (9 * len(n_str)))
    
    for i in range(start, n):
        total = i + sum(map(int, str(i)))
        if total == n:
            print(i)
            break
    else:
        print(0)
        
if __name__ == '__main__':
    main()

File: 백준/Bronze/2231. 분해합/분해합.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    string n_str; cin >> n_str;
    int n = stoi(n_str);
    int start = max(0, n - 9 * (int)n_str.length());

    for (int i = start; i < n;  i++){
        int total = i;
        for(auto c : to_string(i)){
            total += (c-'0');
        }
        if (total == n){
            cout << i;
            return 0;
        }
    }
    cout << 0;
}